src/shogun/io/streaming/ParseBuffer.h

```diff
@@ -220,6 +225,18 @@ template <class T> class CParseBuffer: public CSGObject
 	bool free_vectors_on_destruct;
 };
 
+
+template <class T> void CParseBuffer<T>::init_vector()
+{
+	if (!free_vectors_on_destruct)
+		return;
+	for (int32_t i=0; i<ring_size; i++)
+	{
+		if(ex_ring[i].fv==NULL)
+			ex_ring[i].fv = new T();
```

Comment:
I tried `SG_MALLOC(T, 1);`, it does not work.
Maybe the size/length is wrong. Do you know how can I get the right size given a generic type `T`?